src/main/java/org/manifold/compiler/front/TupleValueExpression.java

```diff
@@ -0,0 +1,60 @@
+package org.manifold.compiler.front;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.Value;
+
+import com.google.common.collect.ImmutableMap;
+
+public class TupleValueExpression extends Expression {
```

Comment:
Would it make more sense to use a `LiteralValueExpression` containing a `TupleValue`?